xtask/src/cli.rs

```diff
@@ -0,0 +1,60 @@
+use std::process::exit;
+
+use anyhow::{anyhow, bail, Context};
+use pico_args::Arguments;
+
+const HELP: &str = "\
+Usage: xtask <COMMAND>
+
+Commands:
+  run-wasm
+
+Options:
+  -h, --help  Print help
+";
+
+pub(crate) struct Args {
+    pub(crate) subcommand: Subcommand,
+}
+
+impl Args {
+    pub fn parse() -> Self {
+        let mut args = Arguments::from_env();
+        log::debug!("parsing args: {args:?}");
+        if args.contains("--help") {
+            eprint!("{HELP}");
+            exit(101);
+        }
+        match (|| -> anyhow::Result<_> {
+            let subcommand = Subcommand::parse(args)?;
+            Ok(Self { subcommand })
+        })() {
+            Ok(this) => this,
+            Err(e) => {
+                eprintln!("{:?}", anyhow!(e));
+                exit(1)
+            }
+        }
+    }
+}
+
+pub(crate) enum Subcommand {
+    RunWasm {
+        // TODO: I'd really like for this to not leak. `Vec<OsString>` plz?
```

Comment:
How does this leak?

Reply:
Oh, derp, I never intend `TODO`s to actually ship. 😅 I'm annoyed here that there's an _abstraction_ leak with `pico_args::Arguments`, not a resource leak.

I think this is fine for now; I'm inclined to just delete the comment, unless you have a better idea.

Reply:
Yeah, that's fine - I don't really see much of an issue with this design.

Reply:
Removed commentary with 0127ed10.